### ISSUE TITLE
[v0.10.0] EXCEPTION: St12out_of_range CInv::GetCommand() : type=-1929379840 unknown type bitcoin in ProcessMessages() 

### ISSUE THREAD

**Issue by bitkevin** [bitcoin/bitcoin#5845]
* OS: Linux 3.13.0-32-generic
* Bitcoind: 0.10.0

```
bitcoin-cli getinfo
{
    "version" : 100000,
    "protocolversion" : 70002,
    "blocks" : 345763,
    "timeoffset" : -1,
    "connections" : 253,
    "proxy" : "",
    "difficulty" : 46684376316.86029053,
    "testnet" : false,
    "relayfee" : 0.00010000,
    "errors" : "EXCEPTION: St12out_of_range       \nCInv::GetCommand() : type=-1929379840 unknown type       \nbitcoin in ProcessMessages()       \n"
}
```

**Comment by bitkevin** [comment]
Again: 

```
bitcoin-cli getinfo
{
    "version" : 100000,
    "protocolversion" : 70002,
    "blocks" : 345951,
    "timeoffset" : 0,
    "connections" : 857,
    "proxy" : "",
    "difficulty" : 46684376316.86029053,
    "testnet" : false,
    "relayfee" : 0.00001000,
    "errors" : "EXCEPTION: St12out_of_range       \nCInv::GetCommand() : type=-2130706431 unknown type       \nbitcoin in ProcessMessages()       \n"
}
```

**Comment by laanwj** [comment]
Internal, non-critical exceptions in the P2P code were ending up in errors.
Was solved in #7114.
